src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1715,6 +1724,81 @@ private void SetPeriodicEvaluation(WorkspaceModel ws)
             return true;
         }
 
+        // Reload's the dummy nodes and serializes the new node information once the package resolves it. 
+        private void ReloadDummyNodes()
+        {
+            JObject dummyNodeJSON = null;
+            WorkspaceModel currentWorkspace = this.CurrentWorkspace;
+
+            if (currentWorkspace == null || string.IsNullOrEmpty(currentWorkspace.FileName))
+            {
+                return;
+            }
+
+            String filePath = Path.GetFullPath(currentWorkspace.FileName);
+
+            CustomNodeManager.AddUninitializedCustomNodesInPath(
+                        Path.GetDirectoryName(filePath),
+                        IsTestMode);
+
+            // Get the dummy node's in the current workspace. 
+            var dummyNodes = currentWorkspace.Nodes.OfType<DummyNode>();
+
+            foreach (DummyNode dn in dummyNodes)
+            {
+                dummyNodeJSON = dn.OriginalNodeContent as JObject;
+
+                if (dummyNodeJSON != null)
+                {
+                    // Deserializing the dummy node and verifying if it is resolved by the downloaded package
+                    NodeModel resolvedNode = WorkspaceModel.GetNodeModelForDummyNode(
+                                                               dummyNodeJSON.ToString(),
+                                                               LibraryServices,
+                                                               EngineController,
+                                                               Scheduler,
+                                                               NodeFactory,
+                                                               IsTestMode,
+                                                               false,
+                                                               CustomNodeManager);
+
+                    if (!(resolvedNode is DummyNode))
+                    {
+                        currentWorkspace.SetNodeViewDataOnResolvedNode(dn, resolvedNode);
```

Comment:
We can do that but we do not want to set these properties if the resolvedNode (that is returned from GetNodeModelForDummyNode function) is also a dummy node. Since we are already checking that in DynamoModel.cs after the nodemodel is returned, it would be one extra step if we want to do it in GetNodeModelForDummyNode().

Reply:
@aparajit-pratap SetNodeViewDataOnResolvedNode() is now being called from inside GetNodeModelForDummyNode method.